app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -1122,6 +1154,79 @@ public void loadAllTypes() {
     }
 
     /**
+     * Loads all the Abilities into memory. Should be done at program start (during splash screen)
+     */
+    public void loadAllAbilities() {
+        // fail safe 0th index of types:
+        abilities[0] = new Ability("None","None");
+        // get the real abilities from the database:
+        for (int i = 1; i < getMaxAbilityID() + 1; i++) {
+            String[] selectionArg = {String.valueOf(i)};
+            Cursor cursor = database.query(ABILITIES_TABLE, null, ABILITIY_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+
+            abilities[i] = new Ability(cursor.getString(cursor.getColumnIndex(NAME)), cursor.getString(cursor.getColumnIndex(DESCRIPTION)));
```

Comment:
Consider adding the abilityID as a private variable to the Ability class (like with Type). This will give easier access to the index in the array. (Same below with EggGroup)